rviz_default_plugins/src/rviz_default_plugins/displays/image/ros_image_texture.cpp

```diff
@@ -319,5 +346,82 @@ void ROSImageTexture::addMessage(sensor_msgs::msg::Image::ConstSharedPtr msg)
   new_image_ = true;
 }
 
+// Function converts src_img from yuv to rgb
+void ROSImageTexture::image_convert_yuv422_to_rgb(uint8_t *dst_img, uint8_t *src_img,
+                                int dst_start_row, int dst_end_row,
+                                int dst_num_cols, uint32_t stride_in_bytes,
+                                std::string src_format) {
+  // generate lookup tables for conversion
+  std::array<int, 256> U_VALUE_FOR_G;
+  std::array<int, 256> V_VALUE_FOR_G;
+  std::array<int, 256> V_VALUE_FOR_R;
+  std::array<int, 256> U_VALUE_FOR_B;
+  // Values generated based on this formula
+  // for converting YUV to RGB
+  // R = Y + 1.403V'
+  // G = Y + 0.344U' - 0.714V'
+  // B = Y + 1.770U'
+  for (int i = 0; i < 256; i++) {
+    U_VALUE_FOR_G[i] = 0.344 * (i - 128);
+    V_VALUE_FOR_G[i] = 0.714 * (i - 128);
+    V_VALUE_FOR_R[i] = 1.403 * (i - 128);
+    U_VALUE_FOR_B[i] = 1.770 * (i - 128);
+  }
```

Comment:
this was done in ros2_publisher to improve the performance so that instead of computing floating point multiplication during runtime then a look up table would be created during initialization and used in runtime. Here you are doing the calculation from within the function which will perform the calculation then access it from the LUT which is not useful.

Either generate the LUT in initialization or get rid of it and make the calulation inline instead